conans/client/generators/__init__.py

```diff
@@ -1,88 +1,51 @@
-    if generator_name == "CMakeToolchain":
-        from conan.tools.cmake import CMakeToolchain
-        return CMakeToolchain
-    elif generator_name == "CMakeDeps":
-        from conan.tools.cmake import CMakeDeps
-        return CMakeDeps
-    elif generator_name == "AutotoolsDeps":
-        from conan.tools.gnu import AutotoolsDeps
-        return AutotoolsDeps
-    elif generator_name == "AutotoolsToolchain":
-        from conan.tools.gnu import AutotoolsToolchain
-        return AutotoolsToolchain
-    elif generator_name == "PkgConfigDeps":
-        from conan.tools.gnu import PkgConfigDeps
-        return PkgConfigDeps
-    elif generator_name == "MSBuildToolchain":
-        from conan.tools.microsoft import MSBuildToolchain
-        return MSBuildToolchain
-    elif generator_name == "MesonToolchain":
-        from conan.tools.meson import MesonToolchain
-        return MesonToolchain
-    elif generator_name == "MesonDeps":
-        from conan.tools.meson import MesonDeps
-        return MesonDeps
-    elif generator_name == "MSBuildDeps":
-        from conan.tools.microsoft import MSBuildDeps
-        return MSBuildDeps
-    elif generator_name == "VCVars":
-        from conan.tools.microsoft import VCVars
-        return VCVars
-    elif generator_name == "IntelCC":
-        from conan.tools.intel import IntelCC
-        return IntelCC
-    elif generator_name == "QbsToolchain" or generator_name == "QbsProfile":
-        from conan.tools.qbs.qbsprofile import QbsProfile
-        return QbsProfile
-    elif generator_name == "VirtualBuildEnv":
-        from conan.tools.env.virtualbuildenv import VirtualBuildEnv
-        return VirtualBuildEnv
-    elif generator_name == "VirtualRunEnv":
-        from conan.tools.env.virtualrunenv import VirtualRunEnv
-        return VirtualRunEnv
-    elif generator_name == "BazelDeps":
-        from conan.tools.google import BazelDeps
-        return BazelDeps
-    elif generator_name == "BazelToolchain":
-        from conan.tools.google import BazelToolchain
-        return BazelToolchain
-    elif generator_name == "XcodeDeps":
-        from conan.tools.apple import XcodeDeps
-        return XcodeDeps
-    elif generator_name == "PremakeDeps":
-        from conan.tools.premake import PremakeDeps
-        return PremakeDeps
-    elif generator_name == "XcodeToolchain":
-        from conan.tools.apple import XcodeToolchain
-        return XcodeToolchain
-    elif generator_name == "NMakeToolchain":
-        from conan.tools.microsoft import NMakeToolchain
-        return NMakeToolchain
-    elif generator_name == "NMakeDeps":
-        from conan.tools.microsoft import NMakeDeps
-        return NMakeDeps
+        raise ConanException(f"Invalid generator '{generator_name}'. "
+                             f"Available types: {', '.join(_generators)}")
+
+    generator_class = _generators.get(generator_name, None)
+    if generator_class is not None:
+        try:
```

Comment:
Maybe integrate the above ``_generators[generator_name]`` in one single ``try-except`` block, with a singly try block for the happy path.

Reply:
Good idea! Thanks :)